### PR TITLE
Added version:wbump task

### PR DESCRIPTION
**Pull request by mchf**
For bumping and creating a helper class for accessing package version within the code.

### CODE REVIEW

File: lib/tasks/version.rake [hunk 1]
```diff
@@ -36,4 +36,31 @@ namespace :version do
...
+
+  desc "Bump package version and create a simple class for acessing the version from sources"
+  task wbump: :bump do
+    puts "Generating helper class"
+
+    Dir.glob("**/lib/*") do |filename|
+      next if !File.directory?(filename)
+
+      module_name = File.basename(filename).downcase
+      target_file = "#{module_name}_version.rb"
```
Comment (lslezak) [inline lib/tasks/version.rake]: The usual convention for a version file is to use `version.rb`, see `find /usr/lib64/ruby/gems/ -name "*version*.rb"`.

File: lib/tasks/version.rake [hunk 2]
```diff
@@ -36,4 +36,31 @@ namespace :version do
       File.write(spec_file, spec)
     end
   end
+
+  desc "Bump package version and create a simple class for acessing the version from sources"
+  task wbump: :bump do
+    puts "Generating helper class"
+
+    Dir.glob("**/lib/*") do |filename|
+      next if !File.directory?(filename)
```
Comment (lslezak) [inline lib/tasks/version.rake]: Name `filename` is a bit misleading here, it actually contains a *directory* name. Rather use `path` or `dirname` in this case.

File: lib/tasks/version.rake [hunk 3]
```diff
@@ -36,4 +36,31 @@ namespace :version do
...
+
+      class_definition = <<-eof
+module Yast
+  class #{module_name.capitalize}Version
+    VERSION = "#{version}".freeze
+  end
+end
+      eof
+
+      open("#{filename}/#{target_file}", "w") do |f|
```
Comment (lslezak) [inline lib/tasks/version.rake]: You can use simpler `File.write("#{filename}/#{target_file}", class_definition)`.

File: lib/tasks/version.rake [hunk 4]
```diff
@@ -36,4 +36,31 @@ namespace :version do
...
+
+      module_name = File.basename(filename).downcase
+      target_file = "#{module_name}_version.rb"
+      target_path = "#{filename}/#{target_file}"
+
+      puts "Writing helper class to: #{target_path}"
+
+      class_definition = <<-eof
+module Yast
+  class #{module_name.capitalize}Version
```
Comment (lslezak) [inline lib/tasks/version.rake]: NP: I'm not sure about the class name, `FoobarVersion::VERSION` might look strange, what about simpler `Foobar::VERSION`?

**Comment by lslezak** [comment]
And what's actually the purpose of the task? Where do you plan to use it?

BTW why `wbump`?

File: lib/tasks/version.rake (hunk 1 above)
Comment (mchf) [inline lib/tasks/version.rake]: well, I just wanted to customize the file name per package

File: lib/tasks/version.rake (hunk 2 above)
Comment (mchf) [inline lib/tasks/version.rake]: ok

File: lib/tasks/version.rake (hunk 4 above)
Comment (mchf) [inline lib/tasks/version.rake]: The class name is based on lib's subdirectory name and I didn't want to produce names like ```Network``` due to potential collisions.

**Comment by mchf** [comment]
wbump == "write bumped version" ... If you have a better idea, I'm open ;-)

**Comment by mchf** [comment]
My main motivation was to write package version into logs to see what version of package customer uses (support config is not always awailable, ...)

**Comment by jreidinger** [comment]
@mchf well, if you use save_y2logs, it contains also output of rpm -qa which have version of all packages, so you do not need support config. save_y2logs is enough.

**Comment by jreidinger** [comment]
In general I do not like this pull request. It adds another duplicite location. We have in past problem that we have two files with version. Now only authoritative one is spec file, so it should be used from there and not having two sources of info again.

If you need access to version in source code, I suggest to add to yast2 class that get used yast module version like `Version.for("yast2-network")`?

**Comment by mchf** [comment]
In earlier times there were ```VERSION``` file which was imported into spec (template). What I did is opposite direction where version is extracted from spec and creates a class. What I don't like it to do something like ```rpm -q``` bcs it doesn't work in installer. Also support config and logs produced by save_y2logs is not always available.

**Comment by lslezak** [comment]
Neither I like the version duplication. And using another non-standard task for version bumping is wrong. I bet somebody will use simple `version:bump` and it will get out of sync...

- In installer you can check the `/.packages.*` files for the versions.
- If the logs are missing you can ask for them :wink: 

I'd suggest just closing the PR without merging, sorry...

**Comment by jreidinger** [comment]
so lets close it

**Pull request closed without merging**